javalin/src/main/java/io/javalin/http/staticfiles/JettyResourceHandler.kt

```diff
@@ -39,17 +41,48 @@ class JettyResourceHandler(val precompressStaticFiles: Boolean = false) : io.jav
         override fun getResource(path: String) = Resource.newClassPathResource("META-INF/resources$path") ?: super.getResource(path)
     }
 
-    inner class PrefixableHandler(private var urlPathPrefix: String) : ResourceHandler() {
+    inner class PrefixableHandler(private var urlPathPrefix: String, private val aliasChecks: List<AliasCheck>) : ResourceHandler() {
         override fun getResource(path: String): Resource {
             val targetResource by lazy { path.removePrefix(urlPathPrefix) }
             return when {
-                urlPathPrefix == "/" -> super.getResource(path) // same as regular ResourceHandler
-                targetResource == "" -> super.getResource("/") // directory without trailing '/'
+                urlPathPrefix == "/" -> getResourceWithAliasChecks(path)!! // same as regular ResourceHandler but with alias checks
+                targetResource == "" -> getResourceWithAliasChecks("/")!! // directory without trailing '/'
                 !path.startsWith(urlPathPrefix) -> EmptyResource.INSTANCE
                 !targetResource.startsWith("/") -> EmptyResource.INSTANCE
-                else -> super.getResource(targetResource)
+                else -> getResourceWithAliasChecks(targetResource)!!
             }
         }
+
+        private fun getResourceWithAliasChecks(path: String) : Resource? {
+            try {
+                var r: Resource? = null
+                if (baseResource != null) {
+                    val caconicalPath = URIUtil.canonicalPath(path)
+                    r = baseResource.addPath(caconicalPath)
+                    if (r != null && r.isAlias && !checkAlias(path, r)) {
+                        return null
+                    }
+                }
+                if ((r == null || !r.exists()) && path.endsWith("/jetty-dir.css")) r = stylesheet
```

Comment:
Where is this code from? What is `/jetty-dir.css`?

Reply:
I just code copied original jetty's implementation of `ResourceHanlder.getResource()` and changed only alias checks. I decided to keep the part with `/jetty-dir.css` (to make sure the new functionallity won't affect the old)